src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -934,6 +936,10 @@ private bool GatherInformationToStartSync()
                         return false; // we have already disposed, nothing to do here
 
                     _lastFlushed = _parent._lastFlushed;
+
+                    if (_lastFlushed.IsValid == false)
```

Comment:
I got NRE on `_lastFlushed.Journal.SetLastReadTxHeader` (`Journal` was `null`, test `RavenDB_10420`) after I added the suggestion to sync data file after recovery. I did it by adding the following at the end of ctor of `StorageEnvironment`:

```
if (isNew == false && _options.ManualSyncing == false)
	SuggestSyncDataFile(); // let's suggest syncing data file after the recovery
```

The issue was that doing a sync _before_ we flushed anything isn't actually valid because `_lastFlushed` wasn't defined in `UpdateJournalStateUnderWriteTransactionLock`. So I decided to not scheduling the sync immediately after loading a database. We'll do it after the first flush.

Reply:
In that case, we need to do a suggested flush, anyway, no?
I don't think it would be triggered until the first commit otherwise, no?

Reply:
The recovery on startup is actually the same as flush - it writes data from journals to data file but it's unsynced stuff. So maybe we should set `JournalApplicator._lastFlushed` at the end of the recovery? Like we do at the end of flush?

Then we can safely suggest data sync after the recovery.

Reply:
Yes, I think we should set that one.